File: solutions/SoldierandBananas.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int k, n, w;
        k = sc.nextInt();
        n = sc.nextInt();
        w = sc.nextInt();
        
        int req = ((w*(w+1))/2)*k;
        int borrow = Math.max(0, req - n);
        System.out.println(borrow);
    }
}

File: solutions/SoldierandBananas.c
#include <stdio.h> 

int main() {
    int k, n, w;
    scanf("%d %d %d", &k, &n, &w);
    int req = ((w*(w+1))/2)*k;
    int borrow = (req > n) ? req - n : 0;
    printf("%d", borrow);
    return(0);
}